bootstrap/cmd/bootstrap/app/server.go

```diff
@@ -482,7 +493,7 @@ func Run(opt *options.ServerOption) error {
 		// if use k8s client-go API, would be quite verbose if we create all resources one by one.
 		// TODO: use API to create ks Components
 		log.Infof("Apply kubeflow Components...")
-		rawCmd := "ks show default | kubectl apply -f -"
+		rawCmd := "ks show default | kubectl apply --validate=false -f -"
```

Comment:
Why are you setting validate to false? This seems dangerous. Where are we supplying invalid config?

Reply:
Without validate=false, kubectl was complaining about not able to find CRD definitions. Skipping validate is a shortcut, let me see if there's other ways to apply.

Reply:
Setting validate to false is highly undesirable. Not sure why that would be necessary 
Also why can't we just do

ks apply? Why do we need to pipe it to kubectl apply?

Reply:
Found the cause of validate bug:
https://github.com/kubeflow/kubeflow/pull/889

ksonnet doesn't support apply from pod but kubectl does; we can remove pipe to kubectl once ksonnet add this feature.